### PR TITLE
Use url route params for assets

### PR DESCRIPTION
**Pull request by grod220**
Main objective: Move asset selection state management into URL

Secondary bug fix, asset selector was not selecting properly:

Screenshot 2024-10-22 at 8 49 17 PM

### CODE REVIEW

File: src/app/v2/trade/[primary]/[numeraire]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      numeraire: data?.find(a => a.symbol === params.numeraire),
+      error,
+      isLoading,
+    }),
+    [data, error, isLoading, params.numeraire, params.primary],
+  );
+};
+
+export default function TradePage({ params }: { params: QueryParams }) {
+  const { primary, numeraire, error, isLoading } = usePathToMetadata(params);
```
Comment (grod220) [inline src/app/v2/trade/[primary]/[numeraire]/page.tsx:30]: The main proposal to pull the symbols from the route params

File: src/app/v2/trade/[primary]/[numeraire]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            {!!error && `Error loading chart: ${String(error)}`}
+            {isLoading && 'Loading...'}
+            {primary && numeraire && <Chart height={512} primary={primary} numeraire={numeraire} />}
+          </Card>
+        </div>
+        <div className='w-full sm:w-1/2 sm:pr-1 lg:w-[336px] lg:pr-0 mb-2'>
+          <Card title='Route Book'>
+            {!!error && `Error loading route book: ${String(error)}`}
+            {isLoading && 'Loading...'}
+            {primary && numeraire && <RouteBook primary={primary} numeraire={numeraire} />}
```
Comment (grod220) [inline src/app/v2/trade/[primary]/[numeraire]/page.tsx:52]: We will have to build loading skeletons soon. Rough work in that direction. But mainly these components no longer take hard coded values and display data from the assets in the url.

File: src/app/v2/trade/[primary]/[numeraire]/page.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import { Chart } from '@/components/chart';
+import { RouteBook } from '@/components/route-book';
+import { Card } from '@penumbra-zone/ui/Card';
+import { useAssets } from '@/shared/state/assets.ts';
+import { useMemo } from 'react';
+
+interface QueryParams {
+  primary: string;
+  numeraire: string;
+}
```
Comment (grod220) [inline src/app/v2/trade/[primary]/[numeraire]/page.tsx:13]: Naming proposal for asset1 and asset2

File: src/components/PairSelector/index.tsx [hunk 4]
```diff
@@ -1,61 +1,104 @@
...
+  if (isMetadata(numeraire)) {
+    numeraireSymbol = numeraire.symbol;
+  } else if (isBalResponse(numeraire)) {
+    numeraireSymbol = getMetadataFromBalancesResponse(numeraire).symbol;
+  } else {
+    throw new Error('unrecognized metadata for numeraireSymbol asset');
+  }
+
+  router.push(`${PagePath.Trade}/${primarySymbol}/${numeraireSymbol}`);
+};
```
Comment (grod220) [inline src/components/PairSelector/index.tsx:58]: If this approach is ok, will move this all to the getters package in /web

File: src/components/PairSelector/index.tsx [hunk 5]
```diff
@@ -1,61 +1,104 @@
...
+
+export interface PairSelectorProps {
+  primary: AssetSelectorValue;
+  numeraire: AssetSelectorValue;
   dialogTitle?: string;
   disabled?: boolean;
 }
 
 export const PairSelector = observer(
```
Comment (grod220) [inline src/components/PairSelector/index.tsx:67]: Perhaps a nextjs wizard may know what the heck this means

Screenshot 2024-10-22 at 8 55 03 PM

Comment (grod220) [reply src/components/PairSelector/index.tsx:67]: Oh wait, fixed it

File: src/shared/state/pair.ts
Comment (grod220) [inline src/shared/state/pair.ts:1]: Killing the mobx store in favor of url params. If useful for composing with other mobx things, we can make a hook to sync this data in the future.

File: src/app/v2/trade/[primary]/[numeraire]/page.tsx (hunk 3 above)
Comment (JasonMHasperhoven) [reply src/app/v2/trade/[primary]/[numeraire]/page.tsx:13]: I don't know what's industry standard, I have also seen: baseAsset/quoteAsset. To me denominator also sounds more standard than numeraire.

File: src/app/v2/trade/[primary]/[numeraire]/page.tsx (hunk 2 above)
Comment (JasonMHasperhoven) [reply src/app/v2/trade/[primary]/[numeraire]/page.tsx:52]: I believe here the RouteBook and Chart, shouldn't accept primary and numeraire as props, but re-use the usePathToMetadata (using next-router within that hook)

File: src/app/v2/trade/[primary]/[numeraire]/page.tsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+// Converts symbol to Metadata
+const usePathToMetadata = (params: QueryParams) => {
+  const { data, error, isLoading } = useAssets();
```
Comment (JasonMHasperhoven) [inline src/app/v2/trade/[primary]/[numeraire]/page.tsx:17]: We should import next-router here to get the params, so that we can use this hook anywhere.

**Comment by grod220** [comment]
Noticing a bit of screen flicker during loading states. This is a symptom of me adding `Loading...` indicators that are plain text. Think in a subsequent PR, we should add proper loading skeletons.

**Review by VanishMax — APPROVED** [review]
LGTM, base and quote assets sound better

**Comment by grod220** [comment]
@JasonMHasperhoven given the downstream deps on this, going to merge before your approval, but happy to makeup for any additional changes you see is needed in a subsequent PR.

**Pull request merged**